google/cloud/bigquery/client.py

```diff
@@ -806,13 +806,16 @@ def get_dataset(
 
     def get_iam_policy(
         self,
-        table: Union[Table, TableReference],
+        table: Union[Table, TableReference, TableListItem, str],
         requested_policy_version: int = 1,
         retry: retries.Retry = DEFAULT_RETRY,
         timeout: float = DEFAULT_TIMEOUT,
     ) -> Policy:
-        if not isinstance(table, (Table, TableReference)):
-            raise TypeError("table must be a Table or TableReference")
+        table = _table_arg_to_table_ref(table, default_project=self.project)
+        if not isinstance(table, (TableReference)):
+            raise TypeError(
```

Comment:
Need unit test that covers this. Possibly this if statement can be deleted if `_table_arg_to_table_ref` already throws?